### DISCUSSION TITLE
Is there a way to specify that a parameter must be an integer value? How would one write the prior for such a value?

### DISCUSSION THREAD

**Discussion by adam-jahraus** [ReactiveBayes/RxInfer.jl#372] [Q&A] [answered]
Is there a way to specify that a parameter must be an integer value? How would one write the prior for such a value?

I'm trying to write a model, in which one of the observed values is an integer . We're doing integer ambiguity resolution, by estimating the number of whole wavelengths between the source and the receiver, we can observe the phase of the incoming signal and get a much more precise distance measurement. 

How would I specify in a model that the value must be an integer? Furthermore, how would I express the prior distribution of that value? It's essentially a discrete, zero mean, normal distribution.

**Comment by adam-jahraus** [comment]
I looked up the name of a discrete normal distribution, and no surprise it's called the Discrete Gaussian Distribution. However, Distributions.jl doesn't appear to have a Discrete Gaussian implemented.  So my question remains, am I able to specify an integer valued parameter in my model?

**Comment by HoangMHNguyen** [comment, marked as answer]
Hi @adam-jahraus! You can try Poisson distribution. This distribution is available in Distributions.jl and RxInfer.jl.

**Reply by adam-jahraus** [reply]
Ah, I see, that's helpful, thanks!
